pkg/platform/runtime/setup/implementations/alternative/resolver.go

```diff
@@ -0,0 +1,21 @@
+package alternative
+
+import (
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/pkg/platform/runtime/artifact"
+)
+
+type Resolver struct {
+	artifactsForNameResolving artifact.Map
+}
+
+func NewResolver(artifactsForNameResolving artifact.Map) *Resolver {
+	return &Resolver{artifactsForNameResolving: artifactsForNameResolving}
+}
+
+func (r *Resolver) ResolveArtifactName(id artifact.ArtifactID) string {
+	if artf, ok := r.artifactsForNameResolving[id]; ok {
+		return artf.Name
+	}
+	return locale.Tl("alternative_unknown_pkg_name", "unknown")
```

Comment:
This locale key is duplicating the one here: https://github.com/ActiveState/cli/pull/2772/files#diff-612cef18aaba9eaeff0a01319b8d2afad73dc05cb80015adce68d8c2bb2d851aR142. Please add it to en-us.yaml and use `locale.T()` to reference it here and there (https://github.com/ActiveState/cli/pull/2772/files#diff-612cef18aaba9eaeff0a01319b8d2afad73dc05cb80015adce68d8c2bb2d851aR142)